### PR TITLE
Feat swaps fees [SW-34]

### PR DESCRIPTION
**Pull request by compojoom**
## What it solves
Swaps now support dynamic fees.

## How to test it
Navigate to the swaps widget - enter different amounts to sell. The applied fee should depend on the fiat amount swapped.

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/features/swap/helpers/data/stablecoins.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  [address: string]: {
+    name: string
+    symbol: string
+    chain: 'gnosis' | 'mainnet' | 'arbitrum' | 'sepolia'
+  }
+} = {
+  '0xdd96b45877d0e8361a4ddb732da741e97f3191ff': {
+    name: 'BUSD Token from BSC',
+    symbol: 'BUSD',
+    chain: 'gnosis',
```
Comment (anxolin) [inline src/features/swap/helpers/data/stablecoins.ts:11]: nit: maybe it is easier to maintain and less redundant if you use a `Record<Chain, Token[]>`?

Comment (compojoom) [reply src/features/swap/helpers/data/stablecoins.ts:11]: yes! The reason I did it like this is that I didn't want to pass the chainId/chainName. 
Also, name, symbol, chain - not necessary actually. We should maybe just go with token addresses.

File: src/features/swap/helpers/fee.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  const isStableCoin = stableCoinAddresses[sellToken?.address] && stableCoinAddresses[buyToken?.address]
+
+  const fiatAmount = Number(orderKind == 'sell' ? sellTokenFiatAmount : buyTokenFiatAmount) || 0
+
+  if (fiatAmount < 100000) {
+    return isStableCoin ? 10 : 35
+  }
+
+  if (fiatAmount < 1000000) {
```
Comment (usame-algan) [inline src/features/swap/helpers/fee.ts:31]:
```suggestion
  if (fiatAmount < 1_000_000) {
```

nit: Its easier to read that way

File: src/features/swap/helpers/fee.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ */
+export const calculateFeePercentageInBps = (orderParams: OnTradeParamsPayload) => {
+  const { sellToken, buyToken, buyTokenFiatAmount, sellTokenFiatAmount, orderKind } = orderParams
+
+  const isStableCoin = stableCoinAddresses[sellToken?.address] && stableCoinAddresses[buyToken?.address]
+
+  const fiatAmount = Number(orderKind == 'sell' ? sellTokenFiatAmount : buyTokenFiatAmount) || 0
+
+  if (fiatAmount < 100000) {
+    return isStableCoin ? 10 : 35
```
Comment (usame-algan) [inline src/features/swap/helpers/fee.ts:28]: Do you think it makes sense to have an enum for these magic numbers? e.g. `StableCoinFee.High`

File: src/features/swap/index.tsx [hunk 4]
```diff
@@ -69,19 +76,63 @@ const SwapWidget = ({ sell }: Params) => {
...
+      alert: palette.warning.main,
+    },
+    partnerFee: {
+      bps: 35,
+      recipient: SWAP_FEE_RECIPIENT,
+    },
+    content: {
+      feeLabel: 'Widget Fee',
+      feeTooltipMarkdown:
+        'The [tiered widget fee](https://help.safe.global/en/articles/178530-how-does-the-widget-fee-work-for-native-swaps) incurred here and charged by CoW DAO for the operation of the CoW Swap Widget is automatically calculated into this quote. It will contribute to a license fee that supports the Safe Community. Neither the Safe Ecosystem Foundation nor Safe (Wallet) operate the CoW Swap Widget and/or CoW Swap.',
```
Comment (usame-algan) [inline src/features/swap/index.tsx:133]: This doesn't seem to work (including formatted links into the tooltip)
Screenshot 2024-07-01 at 11 21 48

**Review by usame-algan — APPROVED** [review]

File: src/features/swap/index.tsx (hunk 4 above)
Comment (compojoom) [reply src/features/swap/index.tsx:133]: Yes, I've reported this to CoW on friday

File: src/features/swap/helpers/fee.ts (hunk 2 above)
Comment (compojoom) [reply src/features/swap/helpers/fee.ts]: good point!

**Pull request merged**